pkg/gcc/rate_controller.go

```diff
@@ -0,0 +1,177 @@
+package gcc
+
+import (
+	"math"
+	"time"
+
+	"github.com/pion/logging"
+)
+
+type rateController struct {
+	log                  logging.LeveledLogger
+	now                  now
+	initialTargetBitrate int
+	minBitrate           int
+	maxBitrate           int
+
+	lastUpdate         time.Time
+	target             int
+	lastState          state
+	latestRTT          time.Duration
+	latestReceivedRate int
+	decreaseEMA        float64
```

Comment:
What does EMA stand for? Maybe worth expanding

Reply:
Exponential moving average, I moved it into a separate struct, this should make it clearer, also how it is calculated.